src/GraphQl/Resolver/Factory/ResolverFactory.php

```diff
@@ -24,20 +25,23 @@ class ResolverFactory implements ResolverFactoryInterface
 {
     public function __construct(
         private readonly ProviderInterface $provider,
-        private readonly ProcessorInterface $processor
+        private readonly ProcessorInterface $processor,
+        private readonly ?PropertyMetadataFactoryInterface $propertyMetadataFactory = null
     ) {
     }
 
     public function __invoke(string $resourceClass = null, string $rootClass = null, Operation $operation = null): callable
     {
         return function (?array $source, array $args, $context, ResolveInfo $info) use ($resourceClass, $rootClass, $operation) {
-            // Data already fetched and normalized (field or nested resource)
-            if ($body = $source[$info->fieldName] ?? null) {
-                return $body;
-            }
+            if (\array_key_exists($info->fieldName, $source ?? [])) {
+                $body = $source[$info->fieldName];
 
-            if (null === $resourceClass && \array_key_exists($info->fieldName, $source ?? [])) {
-                return $body;
+                $propertyMetadata = $rootClass ? $this->propertyMetadataFactory?->create($rootClass, $info->fieldName) : null;
```

Comment:
Do you think that we can use the property metadata from https://github.com/api-platform/core/blob/62263d08c9bfbc50484bd444b33e4c3a8a6bf836/src/GraphQl/Type/FieldsBuilder.php#L356 ? Adding it to the above closure this would prevent from calling the property metadata factory here as this is quite a hot spot.

Reply:
yes sure. added it to the closure and dropped it at other place.